src/validation/statement.rs

```diff
@@ -1484,6 +1488,41 @@ fn validate_assignment_type_sizes<T: AnnotationMap>(
         });
 }
 
+/// Validates if a POU call has the correct amount of arguments. Specifically for functions the
+/// argument count must be greater or equal depending on if a variadic parameter is present whereas for
+/// stateful POUs the argument count can be less since VAR_INPUT and VAR_OUTPUT arguments are optional
```

Comment:
```suggestion
/// Validates if a POU call has the correct number of arguments. Specifically, for functions,
/// the argument count must be equal to the required count unless the interface is variadic,
/// in which case the argument count may be greater than or equal to the required count. For stateful
/// POUs, the argument count can be less than or equal to the required count since VAR_INPUT and 
/// VAR_OUTPUT arguments are optional.
```